### ISSUE TITLE
Delve fails to list sources

### ISSUE THREAD

**Issue by ayalonsivan** [go-delve/delve#3062]
Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

If this is about source listing not showing up (or breakpoints not being
accepted) while running in a container please read our
[FAQ](https://github.com/go-delve/delve/blob/master/Documentation/faq.md#substpath)
first.

1. What version of Delve are you using (`dlv version`)? 1.9.0
2. What version of Go are you using? (`go version`)? 1.17.11
3. What operating system and processor architecture are you using? macOS ARM

What did you do?
=
1. Run `~/go/bin/dlv debug`
2. Type `b main.main`
3. type `c`

What did you expect to see?
=
Breakpoint should stop and list the sources

What did you see instead?
=
Breakpoint stopped but no sources were available
```
(dlv) bt
0  0x000000010e72ce30 in main.main
   at :0
1  0x00000001046b039c in runtime.main
   at runtime/proc.go:255
2  0x00000001046e5e04 in runtime.goexit
   at runtime/asm_arm64.s:1133
```
Running `sources cloudplatform` lists the source file, but isn't showed as a local file:
```
github.com/wiz-sec/wiz/cloudplatform-fetcher/cmd/root.go
```

Doing the same thing on another macOS computer resulted in the sources being shown:
```
> main.main() ./root.go:90 (hits goroutine(1):1 total:1) (PC: 0xcfc6def)
```
and running `sources cloudplatform` listed the file as a local file:
```
/Users/darwish/src/work/wiz/cloudplatform-fetcher/cmd/root.go
```

**Comment by aarzilli** [comment]
You are probably using -trimpath, see the FAQ.

**Comment by ayalonsivan** [comment]
Not that I'm aware of. I just ran `dlv debug`. I don't use an external build system, didn't pass `--build-flags` and don't have any golang related environment variables set other than `GOPRIVATE`. Is there any other way in which `-trimpath` maybe used?

Also, as I said, doing the same thing with the same repo on another computer seems to have worked. Is there any golang configuration file that I'm not aware of?

**Comment by ayalonsivan** [comment]
I ran `go env` and found out that `-trimpath` is indeed configured in `~/Library/Application Support/go/env` so I deleted that flag, cleared the go cache, removed everything from GOPATH and recompiled. 

I ran `dlv debug` again and this time when I type `sources cloudplatform` I do see a local path of the sources. However, when stopping the breakpoint in `main` I still see:

```
(dlv) bt
0  0x000000010c980e30 in main.main
   at :0
1  0x000000010290439c in runtime.main
   at /usr/local/go/src/runtime/proc.go:255
2  0x0000000102939e04 in runtime.goexit
   at /usr/local/go/src/runtime/asm_arm64.s:1133
```

The path is a full path for `asm_arm64` so removing trimpath did something, but delve still cannot find my source files.

**Comment by aarzilli** [comment]
Can you upload the executable file produced by `dlv debug` somewhere?

**Comment by ayalonsivan** [comment]
Unfortunately this is a private company app, so I cannot upload it. I tried reproducing the problem with a simple hello world packages and it doesn't reproduce.
I also found out that after removing `-trimpath` debugging works as expected in each package other than the main package. It's only the main package which shows `at :0` listed as the source.
When I tried building a hello world app with `--build-flags="-trimpath"` source lookup didn't work, but I did see ` at example/hello/root.go:3` as the source of main, contrary to my company's app in which is see `at :0` even with `-trimpath` removed.
Anyway, thank you for all of your help. I considered my problem to be solved at the moment.
